### PR TITLE
add persistent view toggle

### PR DESCRIPTION
**Pull request by SaraVieira**
Creates the list/grid toggle and its persistence in local storage

Screenshot 2020-04-29 at 13 19 23

### CODE REVIEW

File: packages/app/src/app/overmind/namespaces/dashboard/actions.ts [hunk 1]
```diff
@@ -78,6 +86,14 @@ export const searchChanged: Action<{ search: string }> = (
   state.dashboard.filters.search = search;
 };
 
+export const viewModeChanged: Action<{ mode: 'grid' | 'list' }> = (
```
Comment (MichaelDeBoey) [inline packages/app/src/app/overmind/namespaces/dashboard/actions.ts:89]: We don't use an object anymore when there's only 1 key

```suggestion
export const viewModeChanged: Action<'grid' | 'list'> = (
```

File: packages/app/src/app/overmind/namespaces/dashboard/actions.ts [hunk 2]
```diff
@@ -78,6 +86,14 @@ export const searchChanged: Action<{ search: string }> = (
...
+  { state, effects },
+  { mode }
```
Comment (MichaelDeBoey) [inline packages/app/src/app/overmind/namespaces/dashboard/actions.ts:91]:
```suggestion
  mode
```

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/4009/commits/91af0d88d7bccc83b63e17a32d166b350d983559 is at https://pr4009.build.csb.dev/s/new.

File: packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <>
+      <Menu>
+        <Menu.Button>
+          {viewMode === 'grid' ? <GridIcon /> : <ListIcon />}
+        </Menu.Button>
+        <Menu.List>
+          {STATES.map(viewState => (
+            <Menu.Item
+              field={viewState.key}
+              onSelect={() => viewModeChanged({ mode: viewState.key })}
```
Comment (MichaelDeBoey) [inline packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx:39]:
```suggestion
              onSelect={() => viewModeChanged(viewState.key)}
```

File: packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import React, { FunctionComponent } from 'react';
+import { useOvermind } from 'app/overmind';
+import { Text, Menu, Stack } from '@codesandbox/components';
+import { GridIcon, ListIcon } from './icons';
+
+const STATES: { name: string; key: 'grid' | 'list'; icon: any }[] = [
```
Comment (MichaelDeBoey) [inline packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx:6]:
```suggestion
const STATES: { name: string; key: 'grid' | 'list'; icon: ComponentType }[] = [
```

File: packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import React, { FunctionComponent } from 'react';
```
Comment (MichaelDeBoey) [inline packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx:1]:
```suggestion
import React, { ComponentType, FunctionComponent } from 'react';
```

File: packages/app/src/app/overmind/namespaces/dashboard/actions.ts [hunk 6]
```diff
@@ -9,11 +9,19 @@ import {
 import Fuse from 'fuse.js';
 import { OrderBy, sandboxesTypes } from './state';
 
+const VIEW_MODE_DASHBOARD = 'VIEW_MODE_CODESANDBOX_DASHBOARD';
```
Comment (siddharthkp) [inline packages/app/src/app/overmind/namespaces/dashboard/actions.ts:12]: this is scoped to the domain, so we can skip the company name :)

Comment (SaraVieira) [reply packages/app/src/app/overmind/namespaces/dashboard/actions.ts]: Done :)

File: packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx (hunk 4 above)
Comment (SaraVieira) [reply packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx:6]: I don't really see the sue in adding that, it has never helped me really :/

File: packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx (hunk 5 above)
Comment (SaraVieira) [reply packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx:1]: I don't really see the sue in adding that, it has never helped me really :/

File: packages/app/src/app/overmind/namespaces/dashboard/actions.ts (hunk 1 above)
Comment (SaraVieira) [reply packages/app/src/app/overmind/namespaces/dashboard/actions.ts:89]: This was never a standard, I perfer to send in keys, its easier to maintain in the future

File: packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx (hunk 4 above)
Comment (MichaelDeBoey) [reply packages/app/src/app/pages/NewDashboard/Components/Filters/ViewOptions/index.tsx:6]: Using `any` is an anti-pattern and `ComponentType` is the correct type here

File: packages/app/src/app/overmind/namespaces/dashboard/actions.ts (hunk 1 above)
Comment (MichaelDeBoey) [reply packages/app/src/app/overmind/namespaces/dashboard/actions.ts:89]: @christianalfoni & @CompuIves have commented this in one of my refactor PRs (can't find it back tho) that we should migrate to no object over time when there's only 1 key.

**Pull request merged**